learn/advanced/filtering_and_faceted_search.md

```diff
@@ -45,29 +33,37 @@ Suppose you have a collection of movies containing the following fields:
+If you want to filter results based on the `director` and `genres` attributes, you must add them to the `filterableAttributes` list:
 
 <CodeSamples id="faceted_search_update_settings_1" />
 
-## Using filters
+**This step is mandatory and cannot be done at search time**  because Meilisearch needs to prepare an index before you can use filters. Updating `filterableAttributes` may take a significant amount of time depending on your dataset size.
+
+::: note
+By default, `filterableAttributes` is empty. Filters do not work without first explicitly adding attributes to the `filterableAttributes` list.
+:::
+
+Filters work with numeric and string values. Empty fields or fields containing an empty array will be ignored.
 
-Once you have configured `filterableAttributes`, you can start using [the `filter` search parameter](/reference/api/search.md#filter). Search parameters are added to at search time, that is, when a user searches your dataset.
+Filters do not work with [`NaN`](https://en.wikipedia.org/wiki/NaN) and infinite values such as `inf` and `-inf` as they are [not supported by JSON](https://en.wikipedia.org/wiki/JSON#Data_types). It is possible to filter infinite and `NaN` values if you parse them as strings, except when handling [`_geo` fields](/learn/advanced/geosearch.md#preparing-documents-for-location-based-search).
+
+## Filter basics
+
+Once you have configured `filterableAttributes`, you can use [the `filter` search parameter](/reference/api/search.md#filter). The `filter` search parameter refines search results by selecting documents matching the given filter and running the search query only on those documents.
+
+::: warning
+[Synonyms](/learn/configuration/synonyms.md) don't apply to filters. Meaning, if you have `SF` and `San Francisco` set as synonyms, filtering by `SF` and `San Francisco` will show you different results.
+:::
 
-`filter` expects a **filter expression** containing one or more **conditions**. A filter expression can be written as a string, as an array, or as a mix of both.
+`filter` expects a **filter expression** containing one or more **conditions**. A filter expression can be written as a string, array, or as a mix of both.
 
 ### Conditions
 
 Conditions are a filter's basic building blocks. They are always written in the `attribute OPERATOR value` format, where:
 
 - `attribute` is the attribute of the field you want to filter on
-- `OPERATOR` is the comparison operator and can be `=`, `!=`, `>`, `>=`, `<`, or `<=`
-- `value` is the value condition for the filter
-
-::: note
-`>`, `>=`, `<`, and `<=` only operate on numeric values and will ignore all other types of values.
-
-When operating on strings, `=` and `!=` are **case-insensitive**.
-:::
+- `OPERATOR` can be `=`, `!=`, `>`, `>=`, `<`, `<=`, `TO`, `EXISTS`, `IN`, `NOT`, `AND`, or `OR`
+- `value` is the value the `OPERATOR` should look for in the `attribute`
 
 #### Examples
```

Comment:
In the above example the genres were capitalized. Perhaps this is a good spot to mention that filters on strings are case-insensitive?

Also, could you remove "Note that" on line 76? It's unnecessary. (_Note that_ I would love to create a Vale rule for this 😉 )

Reply:
>Perhaps this is a good spot to mention that filters on strings are case-insensitive?

I mentioned this under the equality and inequality operator headings.